asulibraries/islandora-repo
# Feedback webform extras module

This feature branch includes a new module AND a change to the configuration for webform's "feedback" form. So, pull in the new code and import ./config/sync/webform.webform.feedback.yml -- and then install the `asu_webform_extras` module by running `drush pm-enable asu_webform_extras`. 

**IMPORTANT STEP:** Configure the feedback form to use your email address for testing purposes. This can be done at the following URL: 

Finally, clear the cache.

A prerequisite to testing this locally would be to have SMTP email configured.

The change is to display the item and collection for which the feedback is being provided. The $_GET variables from the Feedback button (link) already provide `item` and `collection` values. _None of the other $_GET values are needed for this feature branch, so it is possible that the feedback button link could be made simpler -- if this is the only place that uses that._

To test:
1. after importing the config, installing the `asu_webform_extras` module, and clearing the cache
2. navigate to a test item (collection page, item, complex object child page) and click the [Feedback] button 
3. from this feedback page, notice that the breadcrumbs should be: HOME / **top parent collection** / **(sub-collection)** / **(complex object parent)** / **(item)** / Feedback -- the links with parentheses around them are optional depending on the actual relationships for the item or collection that is being used.
4. Notice that there is a read-only output near the bottom of the form for "Feedback on" that should provide links to the item and/or collection
5. Submit some feedback for the test item
6. inspect the email that should be delivered to your email address to see that there is a link to the item and/or collection

NOTE: This webform Feedback form is the copy of the instance that is intended to run on KEEP PROD, so this may need to be moved into a different config/ subfolder.
